gammapy/cube/tests/test_cube_pipe.py

```diff
@@ -0,0 +1,121 @@
+                               **def_image)
+    return empty_cube
+
+
+@requires_dependency('reproject')
+@requires_data('gammapy-extra')
+def test_cube_pipe(tmpdir):
+    tmpdir = str(tmpdir)
+    from subprocess import call
+    outdir = tmpdir
+    outdir2 = outdir + '/background'
+
+    cmd = 'mkdir -p {}'.format(outdir2)
+    print('Executing: {}'.format(cmd))
+    call(cmd, shell=True)
+
+    ds = DataStore.from_dir("$GAMMAPY_EXTRA/datasets/hess-crab4-hd-hap-prod2")
+    ds.copy_obs(ds.obs_table, tmpdir)
+    data_store = DataStore.from_dir(tmpdir)
+
+    # Create the background model from the 4 Crab observations
+    bgmaker = OffDataBackgroundMaker(data_store, outdir=outdir2)
+
+    bgmaker.select_observations(selection='all')
+    bgmaker.group_observations()
+    bgmaker.make_model("2D")
+    bgmaker.save_models("2D")
+
+    fn = outdir2 + '/group-def.fits'
+
+    # New hdu table that contains the link to the background model
+    hdu_index_table = bgmaker.make_total_index_table(
+        data_store=data_store,
+        modeltype='2D',
+        out_dir_background_model=outdir2,
+        filename_obs_group_table=fn
+    )
+
+    fn = outdir + '/hdu-index.fits.gz'
+    hdu_index_table.write(fn, overwrite=True)
+
+    center = SkyCoord(83.63, 22.01, unit='deg').galactic
+    offset_band = Angle([0, 2.49], 'deg')
+
+    ref_cube_images = make_empty_cube(image_size=250, energy=[Energy(0.5, "TeV"), Energy(100, "TeV"), 5], center=center)
+    ref_cube_exposure = make_empty_cube(image_size=250, energy=[Energy(0.1, "TeV"), Energy(120, "TeV"), 80],
+                                        center=center, data_unit="m2 s")
+    ref_cube_skymask = make_empty_cube(image_size=250, energy=[Energy(0.5, "TeV"), Energy(100, "TeV"), 5],
+                                       center=center)
+
+    data_store = DataStore.from_dir(tmpdir)
+
+    refheader = ref_cube_images.sky_image_ref.to_image_hdu().header
+    exclusion_mask = SkyMask.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')
+    exclusion_mask = exclusion_mask.reproject(reference=refheader)
+    ref_cube_skymask.data = np.tile(exclusion_mask.data, (5, 1, 1))
+    # Pb with the load psftable for one of the run that is not implemented yet...
+    data_store.hdu_table.remove_row(14)
+
+    # Cube Analysis
+    cube_maker = StackedObsCubeMaker(empty_cube_images=ref_cube_images, empty_exposure_cube=ref_cube_exposure,
+                                     offset_band=offset_band, data_store=data_store, obs_table=data_store.obs_table,
+                                     exclusion_mask=ref_cube_skymask, save_bkg_scale=True)
+    cube_maker.make_images(make_background_image=True, radius=10.)
+
+    assert_allclose(cube_maker.counts_cube.data.sum(), 4898.0, atol=3)
+    assert_allclose(cube_maker.bkg_cube.data.sum(), 4259.818621739171, atol=3)
+    cube_maker.significance_cube.data[np.where(np.isnan(cube_maker.significance_cube.data))] = 0
```

Comment:
There's an option `equal_nan=True` for `assert_allclose`, which could be used instead of fixing the values manually.